src/sas/sasgui/perspectives/fitting/basepage.py

```diff
@@ -1642,7 +1642,13 @@ def _reset_parameters_state(self, listtorestore, statelist):
                         item_page[2].SetValue(item_page_info[2])
                         if item_page[2].__class__.__name__ == "ComboBox":
                             if item_page_info[2] in self.model.fun_list:
-                                fun_val = self.model.fun_list[item_page_info[2]]
+                                # to fix: fun_list is not a dictionary, but a tuple
+                                # so the following line (commented) will fail
+                                # fun_val = self.model.fun_list[item_page_info[2]]
+                                # I guess the following should work, but 
+                                # could not test as I don't know when this
+                                # part is used by SasView.
+                                fun_val = self.model.fun_list.index(item_page_info[2])
```

Comment:
Try saving a project with spherical_sld (several layers with different functions in each layer) and see if it can be reloaded.  

Try fitting and see if the function boxes remain unchanged after the fit (both the value and the list of available functions).  

Try creating a plugin model with a choice list and see that you can edit the plugin without the GUI doing silly things.  E.g.:
```
parameters = [
   ["option", "", 0, [["option 0", "option 1", "option 2"]], "", "function selection"],
]

def Iq(q, option):
    return q**option
```

Reply:
Tests OK. I could fit, change repeatedly the number of shells, save and load and create a plugin model with a combo box, and did not observe any anomalous behaviour.